### PR TITLE
Optimise running of EMR jobs

### PR DESCRIPTION
**Pull request by andyphelps**
* Add removal of tiff and png files when not local running
* Update README.md for local deployment
* Update default cluster to 10 nodes and m4.10xlarge for master

### CODE REVIEW

File: ais-batch-lambda/src/main/java/uk/gov/ukho/aisbatchlambda/JobsProvider.java [hunk 1]
```diff
@@ -5,142 +5,148 @@
...
-  private static final Double DEGREES_AT_EQUATOR_FOR_1KM_RESOLUTION = 0.008983031;
-  private static final long DISTANCE_INTERPOLATION_THRESHOLD = 30000L;
-  private static final long TIME_INTERPOLATION_THRESHOLD = 6L * 60 * 60 * 1000;
+    private static final Double DEGREES_AT_EQUATOR_FOR_1KM_RESOLUTION = 0.008983031;
+    private static final long DISTANCE_INTERPOLATION_THRESHOLD_30KM = 30L * 1000;
+    private static final long DISTANCE_INTERPOLATION_THRESHOLD_100KM = 500L * 1000;
+    private static final long DISTANCE_INTERPOLATION_THRESHOLD_500KM = 100L * 1000;
+    private static final long TIME_INTERPOLATION_THRESHOLD_6HR = 6L * 60 * 60 * 1000;
+    private static final long TIME_INTERPOLATION_THRESHOLD_18HR = 18L * 60 * 60 * 1000;
+    private static final long TIME_INTERPOLATION_THRESHOLD_48HR = 48L * 60 * 60 * 1000;
```
Comment (JoelClemence) [inline ais-batch-lambda/src/main/java/uk/gov/ukho/aisbatchlambda/JobsProvider.java]: ```60 * 60 * 1000``` could be parameterised?

File: build.gradle.kts [hunk 2]
```diff
@@ -11,7 +11,7 @@ allprojects {
     apply { plugin("com.diffplug.gradle.spotless") }
 
     group = "uk.gov.ukho"
-    version = "1.5-SNAPSHOT"
+    version = "1.6.2-SNAPSHOT"
```
Comment (JoelClemence) [inline build.gradle.kts]: Maybe just ```1.6-SNAPSHOT```?

File: spark-job/README.md [hunk 3]
```diff
@@ -10,3 +10,6 @@ Outputting the interpolated points as a TSV can be done using:
     }
     .saveAsTextFile("interpolated.csv")
       
+Run locally with:
+
+~/spark/spark-2.4.3-bin-hadoop2.7/bin/spark-submit --class uk.gov.ukho.ais.rasters.AisToRaster spark-job/build/libs/spark-job-1.6-SNAPSHOT-all.jar -i /home/andy/Downloads/boundaries.csv -l -o . -p monthly-world-1k-Sep-18 -r 0.1 -d 30000 -t 21600000 -s 2018-09-01 -e 2018-09-30
```
Comment (JoelClemence) [inline spark-job/README.md]: Is there notes on installing spark?

File: deployment/src/main/terraform/modules/emr_lambda/main.tf [hunk 4]
```diff
@@ -33,14 +33,14 @@ resource "aws_lambda_function" "emr_lambda" {
       JOB_LOCATION = "s3://${var.jobs_bucket_name}/${var.spark_job_jar_name}"
       INPUT_LOCATION = "s3://${var.ais_bucket_name}/*.bz2"
       OUTPUT_LOCATION = "${var.heatmap_bucket_name}"
-      INSTANCE_TYPE_MASTER = "m4.4xlarge"
+      INSTANCE_TYPE_MASTER = "m4.10xlarge"
```
Comment (ciaranevans) [inline deployment/src/main/terraform/modules/emr_lambda/main.tf]: Change back to 4x?

File: spark-job/README.md (hunk 3 above)
Comment (ciaranevans) [reply spark-job/README.md]: Can this be formatted like the above snippet? And generalised for different peoples envs?

File: ais-batch-lambda/src/test/java/uk/gov/ukho/aisbatchlambda/JobsProviderTest.java [hunk 5]
```diff
@@ -7,8 +7,12 @@
...
 public class JobsProviderTest {
   private static final Double DEGREES_AT_EQUATOR_FOR_1KM_RESOLUTION = 0.008983031;
-  private static final long DISTANCE_INTERPOLATION_THRESHOLD = 30000L;
-  private static final long TIME_INTERPOLATION_THRESHOLD = 21600000L;
+  private static final long DISTANCE_INTERPOLATION_THRESHOLD_30KM = 30000L;
+  private static final long TIME_INTERPOLATION_THRESHOLD_6HR = 21600000L;
+  private static final long DISTANCE_INTERPOLATION_THRESHOLD_100KM = 100L * 1000;
+  private static final long DISTANCE_INTERPOLATION_THRESHOLD_500KM = 500L * 1000;
+  private static final long TIME_INTERPOLATION_THRESHOLD_18HR = 18L * 60 * 60 * 1000;
+  private static final long TIME_INTERPOLATION_THRESHOLD_48HR = 48L * 60 * 60 * 1000;
```
Comment (eganjs) [inline ais-batch-lambda/src/test/java/uk/gov/ukho/aisbatchlambda/JobsProviderTest.java]: Would like to see some consistency here

File: ais-batch-lambda/src/test/java/uk/gov/ukho/aisbatchlambda/JobsProviderTest.java [hunk 6]
```diff
@@ -24,122 +28,122 @@ public void whenGetJobsThenCorrectJobsReturned() {
...
-                DISTANCE_INTERPOLATION_THRESHOLD,
-                TIME_INTERPOLATION_THRESHOLD,
-                "2018-06-01",
-                "2018-06-30"),
-            tuple(
-                DEGREES_AT_EQUATOR_FOR_1KM_RESOLUTION,
-                "monthly-world-1k-Jul-18",
-                DISTANCE_INTERPOLATION_THRESHOLD,
-                TIME_INTERPOLATION_THRESHOLD,
+                "monthly-world-1k-30km-6hr-Jul-18",
```
Comment (eganjs) [inline ais-batch-lambda/src/test/java/uk/gov/ukho/aisbatchlambda/JobsProviderTest.java]: CAN these be generated, seems very manual...

Comment (ciaranevans) [reply ais-batch-lambda/src/test/java/uk/gov/ukho/aisbatchlambda/JobsProviderTest.java]: Possibly? Think it's better to have them manual here (could generate them in the Scala but seems nasty to do that)

**Pull request merged**